package/kedro_viz/integrations/kedro/telemetry.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2021 QuantumBlack Visual Analytics Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+# or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""`kedro_viz.integrations.kedro.telemetry` helps integrate Kedro Viz with Kedro-Telemetry
+"""
+import hashlib
+import logging
+import socket
+from pathlib import Path
+from typing import Optional
+
+import yaml
+from kedro_telemetry.plugin import _get_heap_app_id, _is_valid_syntax
+
+logger = logging.getLogger(__name__)
+
+
+def get_heap_app_id(project_path: Path) -> Optional[str]:
+    """Return the Heap App ID used for Kedro telemetry if user has given consent."""
+    telemetry_file_path = project_path / ".telemetry"
+    if not telemetry_file_path.exists():
+        return None
+    with open(telemetry_file_path) as telemetry_file:
+        telemetry = yaml.safe_load(telemetry_file)
+        if _is_valid_syntax(telemetry) and telemetry["consent"]:
+            return _get_heap_app_id()
+    return None
+
+
+def get_heap_identity() -> Optional[str]:  # pragma: no cover
```

Comment:
Ideally there should be a function like this in `kedro-telemetry`. I can mock out hashlib and test it's being called but it's not testing anything useful. Adding `no cover` instead and will PR on the telemetry side to expose this logic as a function.